packages/@aws-cdk/core/lib/stage.ts

```diff
@@ -141,6 +143,12 @@ export class Stage extends CoreConstruct {
 
     this._assemblyBuilder = this.createBuilder(props.outdir);
     this.stageName = [this.parentStage?.stageName, id].filter(x => x).join('-');
+
+    if (scope instanceof App || scope instanceof Stage) {
+      for (const tag of Aspects.of(scope as Stage).aspects) {
+        Aspects.of(this).add(tag);
+      }
```

Comment:
Not bad, but this only works for aspects added before `new Stage`. It will not pick up aspects added *after* `new Stage`.

This kind of order dependency is exactly why we can't support this.